include/roboptim/core/function/polynomial.hh

```diff
@@ -72,6 +72,16 @@ namespace roboptim
     value_type applyPolynomial
     (const_vector_ref coeffs, const_argument_ref x) const;
 
+  public:
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Flag representing the Roboptim Function type
+    static const unsigned long flag_;
+
```

Comment:
We **need** this virtual method and this static flag if an extra flag is enabled, else nothing needs to be done. In any case, the user simply needs to call `flag()`, so I don't really see what good having this outside of macro brings us. Macros tend to obfuscate code, however if we have something like `ROBOPTIM_FUNCTION_ADD_FLAG(IS_DIFFERENTIABLE)` that takes care of the static variable and the virtual getter, things wouldn't be that bad, would they?

Also, does the static flag really needs to be public? I think it should be protected, and then the underscore in the name makes sense regarding our naming guidelines (and we have no problem with naming clashes). That would just make things more slippery if using a macro, if the macro itself contains an access modifier like `protected`.